frontend/packages/dev-console/src/utils/__tests__/pipeline-augment.spec.ts

```diff
@@ -328,3 +296,38 @@ describe('Pipeline exists test to determine whether a pipeline is linked to a pi
     expect(pipelineFound).toBeFalsy();
   });
 });
+
+describe('Pipelinerun graph to show the executed pipeline structure', () => {
+  const currentPipeline = pipelineTestData[PipelineExampleNames.SIMPLE_PIPELINE].pipeline;
+  const testPipelineRun =
+    pipelineTestData[PipelineExampleNames.SIMPLE_PIPELINE].pipelineRuns[DataState.SUCCESS];
+
+  it('expect to return null, if pipelinerun does not have a status field', () => {
+    const plrWithoutPipelineSpec = _.omit(testPipelineRun, ['status']);
+    const executedPipeline = getPipelineFromPipelineRun(plrWithoutPipelineSpec);
+    expect(executedPipeline).toEqual(null);
+  });
+
+  it('expect to return null, if pipelinerun does not have pipelineSpec in status field', () => {
+    const executedPipeline = getPipelineFromPipelineRun(
+      _.omit(testPipelineRun, ['status.pipelineSpec']),
+    );
+    expect(executedPipeline).toBe(null);
+  });
+
+  it('expect to return the previous pipeline structure, if pipelinerun has pipelineSpec in status field', () => {
+    const plrWithOneTask = {
+      ...testPipelineRun,
+      status: {
+        pipelineSpec: {
+          ...currentPipeline.spec,
+          tasks: [currentPipeline.spec.tasks[0]],
+        },
+      },
+    };
+    const executedPipeline = getPipelineFromPipelineRun(plrWithOneTask);
+    expect(currentPipeline.spec.tasks).toHaveLength(2);
```

Comment:
This is a weird `expect`... you're not testing data you used to deliver the test of `getPipelineFromPipelineRun` you trimmed it out on 324 to being only the first one. What is this expect doing other than testing you have data that has 2 points (which is a bit odd of a test)?

Reply:
Yeah I Agree, after the pipeline dependancy removal this test wasn't serving the purpose. So i updated it now.